bin/lesson_check.py

```diff
@@ -257,37 +256,56 @@ def read_all_markdown(source_dir, parser):
     {path : {'metadata':yaml, 'metadata_len':N, 'text':text, 'lines':[(i, line, len)], 'doc':doc}}
     """
 
+    result = {}
+    for d in SOURCE_DIRS:
+        dpath = os.path.join(source_dir, d)
+
+        pattern = os.path.join(dpath, '*.md')
+        for filename in glob.glob(pattern):
+            data = read_markdown(parser, filename)
+            if data:
+                result[filename] = data
+
+    return result
+
     all_dirs = [os.path.join(source_dir, d) for d in SOURCE_DIRS]
+    print(all_dirs)
     all_patterns = [os.path.join(d, '*.md') for d in all_dirs]
+    print(all_patterns)
     result = {}
     for pat in all_patterns:
+        print(pat)
         for filename in glob.glob(pat):
             data = read_markdown(parser, filename)
             if data:
+                print(filename)
                 result[filename] = data
-    return result
+#    return result
 
 
 def check_fileset(source_dir, reporter, filenames_present):
     """Are all required files present? Are extraneous files present?"""
 
     # Check files with predictable names.
-    required = [p.replace('%', source_dir) for p in REQUIRED_FILES]
+    required = [os.path.join(source_dir, p) for p in REQUIRED_FILES]
     missing = set(required) - set(filenames_present)
     for m in missing:
         reporter.add(None, 'Missing required file {0}', m)
 
     # Check episode files' names.
     seen = []
-    for filename in filenames_present:
-        if '_episodes' not in filename:
+    for filepath in filenames_present:
```

Comment:
please undo the unnecessary `filename` -> `filepath` change

Reply:
I changed the name of the variable to reflect the fact that it contains more than just the file name. I split it below using `os.path.basename` and I assign the result of that to `fname`.

Reply:
let's keep changes to a minimum here. We can rename variables in a separate PR.

Reply:
What would be your suggestion for variable names then?

Reply:
keep `filename`. `fname` can be `bname` or `basename` in this PR. Again, we can change variable names in a separate PR.

Reply:
Done.

Reply:
thank you!